racksdb/web/ui/src/views/DatacenterDetailsView.vue

```diff
@@ -0,0 +1,80 @@
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http'
+import { ref, onMounted } from 'vue'
+import SearchBar from '@/components/SearchBar.vue'
+import type { Ref } from 'vue'
+import type { Datacenter, DatacenterRoom } from './DatacentersView.vue'
+
+const http = useHttp()
+const datacenters: Ref<Array<Datacenter>> = ref([])
+const datacenterDetails: Ref<Datacenter | undefined> = ref()
+    
+function roomNbRacks(room: DatacenterRoom) {
+    return room.rows.reduce((result, row) => result + row.nbracks, 0)
+}
+
+async function getDatacenters(){
+    try {
+        const resp = await http.get('datacenters')
+        datacenters.value = resp.data as Datacenter[]
+        datacenterDetails.value = datacenters.value.filter((datacenter) => datacenter.name === props.name)[0]
+    } catch (error) {
+        console.error('Error when fetching the data of datacenters', error)
+    }
+}    
+    
+onMounted(() => {
+    getDatacenters()
+})
+   
+const props = defineProps({
+    name: String        
+})
+</script>
+
+<template>
+    <SearchBar 
+        v-if="datacenters.length"
+        viewTitle="Datacenter Details"    
+        searchedItem="datacenter" 
+        :items="datacenters"
+    />
+    
+    <div class="relative overflow-x-auto py-20">
+        <h2 class="text-3xl font-medium flex justify-center capitalize py-16">{{ name }} Datacenter</h2>
+        <table class="w-full text-sm text-left text-gray-500 dark:text-gray-400">
+            <thead class="text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
+                <tr>
+                    <th scope="col" class="px-6 py-3">Name</th>
+                    <th scope="col" class="px-6 py-3">Area (m²)</th>
+                    <th scope="col" class="px-6 py-3">Number of racks</th>
+                    <th scope="col">Access to the room</th>
+                </tr>
+            </thead>
+
+            <tbody v-for="room in datacenterDetails?.rooms" :key="room.name">
```

Comment:
I the whole table should not be be display while datacenterDetails is undefined, instead of relying on optional chaining operator for its body.